### ISSUE TITLE
To provide a custom localization for an Orchard localization is hard to know and undocumented

### ISSUE THREAD

**Issue by jersiovic** [OrchardCMS/OrchardCore#6464]
If you want to provide a custom localization for a string that is currently translated by Orchard it is not going to be easy.
For example I wanted to provide an alternate traslation for this Spanish traslation:

#. reportError("Password", T["Passwords must have at least one non letter or digit character."]);
#: OrchardCore.Users.Core\Services\UserService.cs:218
msgctxt "OrchardCore.Users.Services.UserService"
msgid "Passwords must have at least one non letter or digit character."
msgstr "Las contraseñas deben tener al menos una letra o un carácter dígito."

Following the docs: https://docs.orchardcore.net/en/dev/docs/reference/modules/Localize/
I provide a traslation in all of these paths to try to achive my purpose, but none of them works.

MyOwnModuleWithADependencyOnOrchardUsers/Localization/es.po
/Localization/es.po
/App_Data/Sites/[TenantName]/Localization/es.po
/Localization/OrchardCore.Users.Core/es.po
/Localization/OrchardCore.Users.Core-es.po
/Localization/es/OrchardCore.Users.Core.po  (this one doesn't work cause it is overwritten on the build)

So, what's happening?
Well problem is Orchard build is copying all the orchard .po files  at /Localization/es/ folder on every build. And it is just that location the one with most priority cause .po files at that folder are the last processed. So, there is not a nice way of overriding the transalation there.

You can see it here https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Localization/ModularPoFileLocationProvider.cs#L81

The only way I've managed to translate it has  been providing a .po file with a weird name to be sure it is the las .po file processed:
/Localization/es/Z.po

Don't you think it should be improved? Or at least document this unexpected way of providing an override of a localization?

**Comment by jtkech** [comment]
@jersiovic yes because the one at the root `\Localization` having the highest precedence

Look at `OC.Application.Cms.Targets` it references `OC.Application.Cms.Core.Targets`, the themes and all translations. So you could do it yourself through a project file (or your own meta package) but by only referencing the translation packages that you need.

Another solution is to exclude, in your project file, the translation files that you don't want

    <Target Name="RemovePackageTranslationFiles" BeforeTargets="CopyPackageTranslationFiles">
      <ItemGroup>
        <PackageTranslationFiles Remove="@(PackageTranslationFiles)" Condition="'%(RecursiveDir)' == 'es\'" />
      </ItemGroup>
    </Target>

**Comment by jersiovic** [comment]
Ok, thank you.
Well in my case I'm happy with all the other translation provided with Orchard in that file. So it is ok for me my solution of overriding only one translation. I just added this issue thinking on others that will follow the same path of pain.

Do you agree if I add a PR to documentation explaining the details for overriding traslations?

**Comment by hishamco** [comment]
> Do you agree if I add a PR to documentation explaining the details for overriding traslations?

I don't think anyone will never mind if this will make this clear ;)

**Comment by jtkech** [comment]
@jersiovic 

> Do you agree if I add a PR to documentation explaining the details for overriding traslations?

Yes no problem 👍
